src/data/simulator.py

```
@@ -0,0 +1,417 @@
+    next_purchase_eligible_day: int = 0     # earliest day a new purchase can occur
+
+    for day in range(n_days):
+        month = day // 30
+        current_date = start_date + timedelta(days=day)
+
+        # --- Check churn state ---
+        if churned:
+            break
+
+        # Daily churn roll: convert monthly probability to daily equivalent so that
+        # short-lived intervention windows (7–14 days) actually have a chance to fire
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)